framework/quota/src/org/apache/cloudstack/quota/dao/QuotaTariffDaoImpl.java

```diff
@@ -0,0 +1,131 @@
+        listAllIncludedUsageType.and("onorbefore", listAllIncludedUsageType.entity().getEffectiveOn(), SearchCriteria.Op.LTEQ);
+        listAllIncludedUsageType.and("quotatype", listAllIncludedUsageType.entity().getUsageType(), SearchCriteria.Op.EQ);
+        listAllIncludedUsageType.done();
+    }
+
+    public QuotaTariffVO findTariffPlanByUsageType(final int quotaType, final Date effectiveDate) {
+        return Transaction.execute(TransactionLegacy.USAGE_DB, new TransactionCallback<QuotaTariffVO>() {
+            @Override
+            public QuotaTariffVO doInTransaction(final TransactionStatus status) {
+                List<QuotaTariffVO> result = new ArrayList<>();
+                final Filter filter = new Filter(QuotaTariffVO.class, "updatedOn", false, 0L, 1L);
+                final SearchCriteria<QuotaTariffVO> sc = listAllIncludedUsageType.create();
+                sc.setParameters("onorbefore", effectiveDate);
+                sc.setParameters("quotatype", quotaType);
+                result = search(sc, filter);
+                if (result != null && !result.isEmpty()) {
+                    return result.get(0);
+                } else {
+                    if (s_logger.isDebugEnabled()) {
+                        s_logger.debug("QuotaTariffDaoImpl::findTariffPlanByUsageType: Missing quota type " + quotaType);
+                    }
+                    return null;
+                }
+            }
+        });
+    }
+
+    public List<QuotaTariffVO> listAllTariffPlans() {
+        return Transaction.execute(TransactionLegacy.USAGE_DB, new TransactionCallback<List<QuotaTariffVO>>() {
+            @Override
+            public List<QuotaTariffVO> doInTransaction(final TransactionStatus status) {
+                return listAll();
+            }
+        });
+    }
+
+    public List<QuotaTariffVO> listAllTariffPlans(final Date effectiveDate) {
+        return Transaction.execute(TransactionLegacy.USAGE_DB, new TransactionCallback<List<QuotaTariffVO>>() {
+            @Override
+            public List<QuotaTariffVO> doInTransaction(final TransactionStatus status) {
+                List<QuotaTariffVO> tariffs = new ArrayList<QuotaTariffVO>();
+                final Filter filter = new Filter(QuotaTariffVO.class, "updatedOn", false, 0L, 1L);
+                final SearchCriteria<QuotaTariffVO> sc = listAllIncludedUsageType.create();
+                sc.setParameters("onorbefore", effectiveDate);
+                for (Integer quotaType : QuotaTypes.listQuotaTypes().keySet()) {
+                    sc.setParameters("quotatype", quotaType);
+                    List<QuotaTariffVO> result = search(sc, filter);
+                    if (result != null && !result.isEmpty()) {
+                        tariffs.add(result.get(0));
+                        s_logger.debug("ListAllTariffPlans on or before " + effectiveDate + " quota type " + result.get(0).getDescription() + " , effective Date="
```

Comment:
This log statement should be wrapped in a ``if (s_logger.isDebugEnabled)`` check.